i6_native_ops/fbw/__init__.py

```diff
@@ -1,2 +1,65 @@
 import torch  # needed to find pytorch specific libs
-from .fbw_core import fbw
+from pkg_resources import get_distribution
+
+try:
+    # Package is installed, so ops are already compiled
+    __version__ = get_distribution('i6_native_ops').version
+    from . import fbw_core as core
+except Exception as e:
+    # otherwise try to build locally
+    from torch.utils.cpp_extension import load
+    base_path = os.path.dirname(__file__)
+    core = load(
+        name="fbw_core",
+        sources=[
+            os.path.join(base_path, "fbw_torch.cpp"),
+            os.path.join(base_path, "fbw_op.cu"),
+        ],
+        extra_include_paths=[
+            base_path,
+            os.path.join(base_path, "..", "common")
+        ],
+    )
+
+
+class FastBaumWelchLoss(torch.autograd.Function):
```

Comment:
I am no expert on custom PT ops, but should the fbw implementation not be similar to the fast_viterbi implementation in  #3?

Reply:
The extra thing for fbw is that it backpropagates the gradients automatically. We can just register the output of `fbw_loss` as a loss and RETURNN/PyTorch handles the rest. The fast viterbi procedure does not give us a differentiable function and we're not interested in the gradients, that's why we can just wrap it into a simple function, giving us alignments and scores.

Reply:
>  viterbi [..] not [..] differentiable

yep makes sense 👍